generic_config_updater/patch_sorter.py

```diff
@@ -0,0 +1,965 @@
+            return operation_wrapper.create(op_type, path, value)
+
+        if op_type == OperationType.ADD:
+            return JsonMove._to_jsonpatch_add_operation(diff, current_config_tokens, target_config_tokens)
+
+        raise ValueError(f"OperationType {op_type} is not supported")
+
+    @staticmethod
+    def _get_value(config, tokens):
+        for token in tokens:
+            config=config[token]
+
+        return copy.deepcopy(config)
+
+    @staticmethod
+    def _to_jsonpatch_add_operation(diff, current_config_tokens, target_config_tokens):
+        """
+        Check description of JsonMove class first.
+
+        ADD operation path can refer to elements that do not exist, so to convert JsonMove to JsonPatch operation
+        We need to remove the non-existing tokens from the current_config path and move them to the value.
+
+        Example:
+        Assume Target Config:
+        {
+            "dict1":{
+                "key11": "value11"
+            }
+        }
+        Assume Current Config:
+        {
+        }
+        Assume JsonMove:
+        op_type=add, current_config_tokens=[dict1, key11], target_config_tokens=[dict1, key11]
+
+        Converting this to operation directly is incorrect since the current_config tokens contains 'dict1' which does
+        not exist in Current Config. Instead we convert to:
+        {"op":"add", "path":"/dict1", "value":{"key11": "value11"}}
+        """
+        operation_wrapper=OperationWrapper()
+        path_addressing=PathAddressing()
+
+        # if path refers to whole config, then just create operation
+        if not current_config_tokens:
+            path = path_addressing.create_path(current_config_tokens)
+            value = JsonMove._get_value(diff.target_config, target_config_tokens)
+            return operation_wrapper.create(OperationType.ADD, path, value)
+
+        # Start with getting target-config that match the path all the way to the value in json format
+        # Example:
+        # Assume target-config:
+        # {
+        #     "dict1":{
+        #         "key11": "value11",
+        #         "list12": [
+        #                     "value121",
+        #                     "value122"
+        #                   ]
+        #     },
+        #     "dict2":{
+        #         "key21": "value21"
+        #     }
+        # }
+        # Assume target config tokens:
+        # dict1, list12, 1
+        # filtered_config will be
+        # {
+        #     "dict1":{
+        #         "list12": [
+        #                     "value122"
+        #                   ]
+        #     }
+        # }
+        target_ptr = diff.target_config
+        filtered_config = {}
+        filtered_config_ptr = filtered_config
+        for token_index in range(len(target_config_tokens)):
+            token = target_config_tokens[token_index]
+
+            target_ptr = target_ptr[token]
```

Comment:
what if the token is a string, but it should be treated as an int. can this happen?